optuna/integration/skorch.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any
+
+import optuna
+
+with optuna._imports.try_import() as _imports:
+    from skorch.callbacks import Callback
+    from skorch.net import NeuralNet
+
+if not _imports.is_successful():
+    Callback = object  # NOQA
+    NeuralNet = object  # NOQA
+
+
+class SkorchPruningCallback(Callback):
+    """Skorch callback to prune unpromising trials.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        monitor:
+            An evaluation metric for pruning, e.g. ``val_loss`` or
+            ``val_acc``. The metrics are obtained from the returned dictionaries,
+            i.e., ``net.histroy``. The names are thus depend on how this dictionary
+            is formatted.
```

Comment:
When ``val_loss`` is used, the user should specify `direction=minimize` in `optuna.create_study`. On the other hand, when ``val_acc`` is used, the user should specify ``direction=maximize` in `optuna.create_study`. I think it is kind for users to provide this information here.

Reply:
That sounds a bit too much to me, and I don't think that kind comment should be on some other place like https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html?highlight=direction#optuna.study.Study.direction

Reply:
I think the more helpful the documentation is, the better for users. I would like to know other reviewers' opinions.

Reply:
Okay, if that's the case, we should file some issues for some pruning callbacks that doesn't convey that information and label them "contribution-welcome".

Reply:
@toshihikoyanase what's your take here?

Reply:
IMO, it is a bit verbose if we add the comments about `direction` here because the other pruning callbacks such as `KerasPruningCallback` and `XGBoostPruningCallback` also have the metrics similar to `monitor`.

So, I'd like to place such a comment in the reference of direction (`Study.direction` or `create_study`) or [tutorial](https://optuna.readthedocs.io/en/stable/tutorial/pruning.html#activating-pruners) of pruning.